core/crypto/ecdsa/ecdsa_provider_impl.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/ecdsa/ecdsa_provider_impl.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::crypto, EcdsaProviderImpl::Error, e) {
+  using E = kagome::crypto::EcdsaProviderImpl::Error;
+  switch (e) {
+    case E::VERIFICATION_FAILED:
+      return "Internal error during ecdsa signature verification";
+    case E::SIGN_FAILED:
+      return "Internal error during ecdsa signing";
+  }
+  return "Unknown error in ecdsa provider";
+}
+
+namespace kagome::crypto {
+
+  EcdsaProviderImpl::EcdsaProviderImpl()
+      : provider_{std::make_shared<Libp2pEcdsaProviderImpl>()} {}
+
+  EcdsaProviderImpl::EcdsaProviderImpl(
+      std::shared_ptr<Libp2pEcdsaProviderImpl> provider)
```

Comment:
Hm, why does it depend on an implementation rather than an interface? Kinda breaks the dependency inversion.

Reply:
Fixed